src/os/bluestore/BlueFS.cc

```diff
@@ -1315,7 +1561,39 @@ void BlueFS::sync_metadata()
       p->commit_finish();
     }
   }
-  _maybe_compact_log();
+  // check whether we need to compact the log or not
+  uint64_t current = log_writer->file->fnode.size;
+  uint64_t expected = _estimate_log_size();
+  float ratio = (float)current / (float)expected;
+  dout(10) << __func__ << " current " << current
+	   << " expected " << expected
+	   << " ratio " << ratio << dendl;
+
+  if (current < g_conf->bluefs_log_compact_min_size ||
+      ratio < g_conf->bluefs_log_compact_min_ratio) {
+    lock.unlock();
+    return;
```

Comment:
why duplicate maybe_compact_log?  if the locking is different, turn it into a helper like _should_compact_log(() or something